### PR TITLE
Add shift days attribute for Doctor class 

### LINKED ISSUE #93: Add shift days for Doctors

### PR DESCRIPTION
**Pull request by leezhanpeng**
closes #93.

Main changes:
- `ShiftDays` class is now available for use.
- `Doctor` class now has a `ShiftDays` attribute.
- Ui is now ready to refer to `Doctor`'s `ShiftDays` when displaying shift days.
- Initialising of MediSync previously instantiates contact list with `Person` class. It has now been updated to the correct `Doctor`/`Patient` classes.

### CODE REVIEW

**Review by Seonlo99 — APPROVED** [review]
LGTM.

File: src/main/java/seedu/address/model/person/Doctor.java [hunk 1]
```diff
@@ -9,24 +10,74 @@
...
     }
 
     /**
      * Every field must be present and not null, except affiliationHistory.
      */
     public Doctor(Name name, Phone phone, Email email, Set<Affiliation> affiliations) {
         super(name, phone, email, new Role("Doctor"), affiliations);
+
+        // Instantiation of Doctor should always begin with empty shift days.
+        shiftDays = new ShiftDays();
```
Comment (EricXiong420) [inline src/main/java/seedu/address/model/person/Doctor.java:33]: Important to take note of when implementing further shift changes, thanks for noting in a comment!

File: src/main/java/seedu/address/model/person/ShiftDays.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+
+    /**
+     * Modifies the set of shift days.
+     *
+     * @param shiftDays The set of integers representing the shift days.
+     */
+    public ShiftDays modifyShiftDays(Set<Integer> shiftDays) {
+        this.shiftDays.clear();
+        this.shiftDays.addAll(shiftDays);
+        return this;
```
Comment (EricXiong420) [inline src/main/java/seedu/address/model/person/ShiftDays.java:78]: Works as a container for a Set of Integers, so the object always remains the same while the set can change. Also possible to implement as immutable object with factory method, was wondering why this approach was chosen?

File: src/main/java/seedu/address/model/person/Doctor.java [hunk 3]
```diff
@@ -9,24 +10,74 @@
...
+                && getAffiliations().equals(otherDoctor.getAffiliations())
+                && getAffiliationHistory().equals(otherDoctor.getAffiliationHistory())
+                && shiftDays.equals(otherDoctor.getShiftDays());
+    }
+
+    @Override
+    public int hashCode() {
+        // use this method for custom fields hashing instead of implementing your own
+        return Objects.hash(getName(), getPhone(), getEmail(), getRole(), getAffiliations(),
+                getAffiliationHistory(), shiftDays);
```
Comment (EricXiong420) [inline src/main/java/seedu/address/model/person/Doctor.java:73]: Code coverage can be addressed here by adding another line in tests using the hashCode method

**Review by EricXiong420 — APPROVED** [review]
A few comments on certain lines, and code coverage, but otherwisw LGTM can go ahead and merge!

**Pull request merged**